lib/core/notifier.js

```diff
@@ -277,30 +276,24 @@ class NotifierController {
    * @param {Request} request
    * @return {Promise}
    */
-  notifyDocumentUpdate (request) {
-    let
-      cacheId,
-      matchedRooms,
-      updatedDocument;
-
-    const { index, collection, _id } = request.input.resource;
-
-    // @todo do not make another get request, use the _source returned by update request
-    return this.storageEngine.get(index, collection, _id)
-      .then(result => {
-        updatedDocument = result;
-        matchedRooms = this._test(request, result._source, result._id);
-        cacheId = getCachePrefix(request) + updatedDocument._id;
-
-        const updatedFields = Object.keys(request.input.body)
-          .filter(_updatedFields => _updatedFields !== '_kuzzle_info');
-
-        return this.notifyDocument(matchedRooms, request, 'in', 'update', {
-          _id: updatedDocument._id,
-          _source: updatedDocument._source,
-          _updatedFields: updatedFields
-        });
-      })
+  notifyDocumentUpdate (request, updatedDocument) {
```

Comment:
You should use `async/await` instead of promise chain in this method.  
We try to progressively get ride of unnecessary promise chain